runtime/src/comm/gasnet/comm-gasnet-ex.c

```diff
@@ -946,8 +991,37 @@ int chpl_comm_run_in_lldb(int argc, char* argv[], int lldbArgnum, int* status) {
   return 0;
 }
 
+static void start_gasnet_progress_threads(void) {
+  unsigned int count;
+  const gex_ProgressThreadInfo_t *info;
+  GASNET_Safe(gex_System_QueryProgressThreads(myclient, &count, &info, 0) );
+  for (int i = 0; i < count; i++) {
+
+    // Don't create an internal receive thread if there is an external one
+    if (pollingRequired &&
+        (info[i].gex_thread_roles == GEX_THREAD_ROLE_RCV)) {
```

Comment:
@jhh67 The line above tests for *equality* to a value which is documented as being just one bit among others which may be set.  Currently, the GASNet ibv-conduit implementation does not set both `GEX_THREAD_ROLE_RCV` and `GEX_THREAD_ROLE_SND`, but in the future it could.
Dan and I discussed this logic and believe the current test for equality is probably the correct one because *if* in the future there was a thread serving both the `RCV` and `SND` roles, you would want it to run for the `SND` activity.  However, we don't know if that was your intent or just the way things worked out.  So, an explanatory comment here would be useful.
TL;DR:  Request something like ` // Test for equality ensures a combination RCV+SND thread will still run`